Automattic/woocommerce-payments
# Fix Multi-Currency switcher support in WooPay

Context: p1684449926045609/1684445728.311959-slack-C022WMN88KG

#### Changes proposed in this Pull Request

* Remove the requirement for `REST_REQUEST` constant in Store API remote requests.
    * Note: Oddly, not all Store API REST requests have that constant set to `true`.
* Set WooPay session handler directly in the `woocommerce_session_handler` hook.
    * Similar to the `REST_REQUEST` constant, some store API requests didn't call the `rest_request_before_callbacks` hook, and this was preventing the Multi-currency feature from retrieving the currency from an existing guest session.

#### Testing instructions

1. Ensure WooPay is eligible and enabled.
2. Enable Multi-Currency + the currency switcher from **WC > Settings > Multi-Currency**.
3. Enable multiple currencies, for example: USD, JPY, EUR.
4. As a guest shopper use the currency switcher to change the currency to JPY.
5. Navigate to the checkout page and enter a valid WooPay email + OTP.
6. From the WooPay checkout, ensure the currency is displayed correctly.
7. In a separate tab, go to the merchant store and change the currency to, "USD", for example.
8. Refresh the WooPay checkout tab and ensure the currency, totals and currency symbol are set to USD.
9. Checkout with WooPay.
10. Ensure the charge is correctly processed for the chosen currency.
11. Logout from WooPay and repeat steps 4-10 as a logged in shopper.

-------------------


**Post merge**
